sycl/include/sycl/detail/pi.h

```diff
@@ -2021,6 +2043,121 @@ pi_result piextEnqueueDeviceGlobalVariableRead(
     size_t count, size_t offset, void *dst, pi_uint32 num_events_in_wait_list,
     const pi_event *event_wait_list, pi_event *event);
 
+///
+/// Virtual memory
+///
+
+/// API for getting information about the minimum and recommended granularity
+/// of physical and virtual memory.
+///
+/// \param context is the context to get the granularity from.
+/// \param device is the device to get the granularity from.
+/// \param param_name is the type of query to perform.
+/// \param param_value_size is the size of the result in bytes.
+/// \param param_value is the result.
+/// \param param_value_size_ret is how many bytes were written.
+pi_result
+piextVirtualMemGranularityGetInfo(pi_context context, pi_device device,
+                                  pi_virtual_mem_granularity_info param_name,
+                                  size_t param_value_size, void *param_value,
+                                  size_t *param_value_size_ret);
+
+/// API for creating a physical memory handle that virtual memory can be mapped
+/// to.
+///
+/// \param context is the context within which the physical memory is allocated.
+/// \param device is the device the physical memory is on.
+/// \param mem_size is the size of physical memory to allocate. This must be a
+///        multiple of the minimum virtual memory granularity.
+/// \param ret_physical_mem is the handle for the resulting physical memory.
+pi_result piextPhysicalMemCreate(pi_context context, pi_device device,
+                                 size_t mem_size,
+                                 pi_physical_mem *ret_physical_mem);
```

Comment:
I've made a start on bring these additions into UR which has raised a question.

Should there be a way to pass flags to physical memory object creation?

My thinking here is you could set a default for access mode that applies to the whole range, could potentially reduce overheads in some cases. Also, it may also make sense to have a flag to opt in/out of zeroing the entire memory region.

Reply:
Its probably fine for me to add it to UR without an extra parameter in PI, this would also enable use of `pNext` chains if/when a use case for extending this functionality comes around. L0 & UR actually have a `TBD` flag for context creation to serve a similar purpose, so I'll do a similar thing for this.